### PR TITLE
Feature/#182201717-user-roles-setting

### PR DESCRIPTION
**Pull request by Patricknshimiyimana**

### CODE REVIEW

**Review by Placide123 — DISMISSED** [review]

File: src/services/user.service.js [hunk 1]
```diff
@@ -37,3 +37,28 @@ export const createUser = async (req, res) => {
...
+            return res.status(404).json({ message: 'User not found' });
+        }
+
+        if (user.role === 'super_admin') {
+           return res.json({message: "Super admin can not be updated"})
+        }
+        updateUser({ email: email }, { role: role })
+       
+        return res.json({result: "user role updated"})
+    })
```
Comment (uwambajeddy) [inline src/services/user.service.js:62]: try to give status code on the responses

Comment (Patricknshimiyimana) [reply src/services/user.service.js:62]: Done. Thanks

File: src/middlewares/isAdmin.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    const authHeader = req.get('Authorization');
+    if (!authHeader) {
+        return res.status(401).json({message: 'Not authenticated'})
+    }
+    
+    const token = authHeader.split(' ')[1];
+    const userId = jwt.verify(token, process.env.JWT_SECRET).id;
+    const user = await users.findByPk(userId);
+
+    if (user.role !== 'super_admin')  {
```
Comment (uwambajeddy) [inline src/middlewares/isAdmin.js]: we use `super admin` not `super_admin` in database

**Review by Placide123 — DISMISSED** [review]

File: src/services/user.service.js [hunk 3]
```diff
@@ -37,3 +38,59 @@ export const createUser = async (req, res) => {
...
+    const email = req.body.email;
+    const role = req.body.role;
+
+    	users.findOne({ where: { email: email } }).then(user => {
+        console.log(user)
+        if(user === null){
+            return res.status(404).json({ message: 'User not found' });
+        }
+
+        if (user.role === 'super_admin') {
```
Comment (uwambajeddy) [inline src/services/user.service.js]: same here

Comment (Patricknshimiyimana) [reply src/services/user.service.js]: Done.

**Review by uwambajeddy — COMMENTED** [review]
`routes` is outdated we are using `routers` folder

**Comment by Patricknshimiyimana** [comment]
> `routes` is outdated we are using `routers` folder

Done!! I deleted it.

File: src/services/user.service.js [hunk 4]
```diff
@@ -37,3 +38,59 @@ export const createUser = async (req, res) => {
...
+    	users.findOne({ where: { email: email } }).then(user => {
+        console.log(user)
+        if(user === null){
+            return res.status(404).json({ message: 'User not found' });
+        }
+
+        if (user.role === 'super admin') {
+           return res.status(403).json({message: "Super admin can not be updated"})
+        }
+        updateUser({ email: email }, { role: role })
```
Comment (joshwambere) [inline src/services/user.service.js:59]: how do you know it updated successfully

Comment (Patricknshimiyimana) [reply src/services/user.service.js:59]: We are not using user.service.js anymore
The  code is in the controllers

**Pull request merged**